packages/commonwealth/server/util/tokenBalanceCache/tokenBalanceCache.ts

```diff
@@ -283,6 +302,11 @@ export class TokenBalanceCache {
         );
       case BalanceSourceType.CosmosNative:
         return `${options.sourceOptions.cosmosChainId}_${address}`;
+      case BalanceSourceType.CW721:
+        return (
+          `${options.sourceOptions.cosmosChainId}_${address}` +
+          `_${options.sourceOptions.contractAddress}`
```

Comment:
This will break caching! Note the warning comment at the top of this function.

Reply:
Missing test cases for CW721 caching tests.

Reply:
👍  Didn't realize that was needed. Will do.

Reply:
Does the existing "Caching" test actually test TTL? It seems like it just shows an updated balance even though the TTL has passed.

Reply:
The test fetches balances 3 times. The first time it uses `cacheRefresh: true` to fetch a fresh value. The second fetch occurs after a balance transfer (so the actual balance has changed) but before the TBC TTL has expired so it should still return the old balance value. The third fetch occurs after a delay that is longer than the TTL so the TBC should return the new/updated/fresh value.

So yes, the caching test actually tests TTL.

Reply:
I added a caching test. LMK if it needs more.

Reply:
>This will break caching! Note the warning comment at the top of this function. 

fixed